platform/mbed_power_mgmt.h

```diff
@@ -68,18 +68,26 @@ extern "C" {
 void sleep_tracker_lock(const char *const filename, int line);
 void sleep_tracker_unlock(const char *const filename, int line);
 
-#define sleep_manager_lock_deep_sleep()           \
-    do                                            \
-    {                                             \
-        sleep_manager_lock_deep_sleep_internal(); \
-        sleep_tracker_lock(__FILE__, __LINE__);   \
-    } while (0);
+#if defined(__CC_ARM) || (defined(__ARMCC_VERSION) && (__ARMCC_VERSION >= 6010050))
+#define __FILENAME__ __MODULE__
+#elif defined(__GNUC__)
+#define __FILENAME__ (__builtin_strrchr(__FILE__, '/') ? __builtin_strrchr(__FILE__, '/') + 1 : __builtin_strrchr(__FILE__, '\\') ? __builtin_strrchr(__FILE__, '\\') + 1 : __FILE__)
```

Comment:
Doesn't GCC map strchr to __builtin_strrchr automatically?

Reply:
Apparently yes, I just hadn't turned on optimizations when checking.

Reply:
You can use `__BASE_FILE__` for GNU. https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html
`__BASE_FILE__`
This macro expands to the name of the main input file, in the form of a C string constant. This is the source file that was specified on the command line of the preprocessor or C compiler.

Reply:
It tried it, but it doesn't work in this situation, it expands to the full path of the module that is passed into make/compiler so the output is the same as `__FILE__` in this scenario.